src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -177,25 +177,8 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
 
                 if (isCpvmEnabled && !await AreCentralVersionRequirementsSatisfiedAsync())
                 {
-                    restoreTime.Stop();
```

Comment:
Are we running a full restore with this change? 
Looking at the code, the pattern here was the same as the lock file one. And it seems like NU1004 might be facing the same issue. 

It's reminding me of https://github.com/NuGet/Home/issues/8973.

Reply:
No. The full restore is run only when _success == true (https://github.com/NuGet/NuGet.Client/blob/df00340c27f8fb257bf6ec3eee5f60521d56e707/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs#L248)

Reply:
Do you know if NU1004 has the same issue as NU1008?

Reply:
If you run a restore of project that generates a NU1004, does NU1004 get written inside the assets file? 

Some repro steps. 

1. Create a PR project 
1. Add a few dependencies & set RestorePackagesWithLockFile. 
1. Edit a csproj manually adding a new PackageREference or editing the version and run `dotnet restore --locked-mode`. 
1. Check if the NU1004 is in the assets file. 

If it is, then we're good, if it isn't that'd be useful detail to add in NuGet/Home#8973.

Reply:
It does not add the information to the assets file. 
Added the information to the bug mentioned.